src/content/docs/en/guides/middleware.mdx

```diff
@@ -19,11 +19,12 @@ Middleware is available in both SSG and SSR Astro projects.
 
     ```js title="src/middleware.js"
     export function onRequest ({ locals, request }, next) {
-        // intercept response data from a request
-        // optionally, transform the response by modifying `locals`
+        // intercept data from a request
+        // optionally, share data with the rendering process by modifying `locals`
```

Comment:
```suggestion
        // optionally, modify the data stored in `locals`
```
While yours may be the more technically accurate description, I don't think this exactly hits the mark on the helpfulness scale.

"Transform some data" is really clear to a reader that you're going to change the value of something. "Share data with a process" really doesn't convey with the same clarity the idea that middleware is used to change an object that is then available somewhere else.

Can we find a middle ground here? Is it "good enough" just to do the above?

Reply:
I agree that my wording was _too_ technical, but it is in line with the language just above it:

> This `locals` object is forwarded across the request handling process and is available as a property to `APIContext` and `AstroGlobal`.

I think a middle ground would be best. The original "transform the response" is what I thought was misleading since it may not affect the response at all, and there is another place to transform the actual `Response` object (after the `next` call), so it could be confusing.

I like your suggestion, we can go with it. I'd prefer "modify the _properties_" instead of "modify the _data_" just because you can set functions in there, but this is quite minor, WDYT?

Reply:
Totally cool with "modify the properties" !